client/record/cli/sendtx.go

```diff
@@ -45,15 +46,24 @@ func GetCmdSubmitFile(storeName string, cdc *wire.Codec) *cobra.Command {
 			// --onchain-data has a high priority over --file-path
 			if len(onchainData) != 0 {
 				dataSize = int64(binary.Size([]byte(onchainData)))
+				if dataSize >= recordClient.UploadLimitOfOnchain {
+					fmt.Printf("File %s is too large, upload limit is %d bytes.\n", filePath, recordClient.UploadLimitOfOnchain)
+					return err
+				}
 				sum := sha256.Sum256([]byte(onchainData))
-				recordHash = hex.EncodeToString(sum[:])
+				recordHash = hex.EncodeToString(sum[:recordClient.IpfsHashLength/2])
 			} else if len(filePath) != 0 {
 				var fileInfo os.FileInfo
 				if fileInfo, err = os.Stat(filePath); os.IsNotExist(err) {
 					fmt.Printf("File %v doesn't exists, please check correstponding path.\n", filePath)
 					return err
 				}
+
 				dataSize = fileInfo.Size()
+				if dataSize >= recordClient.UploadLimitOfIpfs {
```

Comment:
UploadLimitOfIpfs， IpfsHashLength， these params should be define in the record module instead of cli code. And can be update by the gov process.

Reply:
Got it, good suggestions.